pandas/core/frame.py

```diff
@@ -11350,6 +11352,13 @@ def cov(
         c -0.150812  0.191417  0.895202
         """
         data = self._get_numeric_data() if numeric_only else self
+        dtypes = [blk.dtype for blk in self._mgr.blocks]
+        if any(is_datetime64_any_dtype(d) or is_timedelta64_dtype(d) for d in dtypes):
```

Comment:
i suspect that doing the check this way (instead of checking dtype.kind like you do below) will fail to exclude pyarrow duration dtypes.